Modix.Services/Promotions/PromotionsService.cs

```diff
@@ -105,57 +124,78 @@ public class PromotionsService : IPromotionsService
         }
 
         /// <inheritdoc />
-        public async Task CreateCampaignAsync(ulong subjectId, ulong targetRoleId, string comment)
+        public async Task<CreateCampaignContinuationData> CreateCampaignAsync(ulong subjectId, string comment)
```

Comment:
Instead of having two methods for the two different "steps" of creating a campaign, I would just add a callback delegate as an optional parameter to this method, I.E. `Func<ProposedPromotionCampaign, Task<bool>> confirmDelegate`.

If you do this, make sure and change the `using` scope of the two transactions. I.E. don't `await` that callback inside the transaction scope, because it will ultimately prevent anyone from creating any campaigns, while a proposed campaign has yet to be confirmed or denied. The corollary is that if two users attempt to create a promotion at the same time, for the same user, one of them will get an exception. Oh well.

Also, yeah, in this context, I think an appropriate name for that new model class would be something like `ProposedPromotionCampaignBrief` or `ProposedPromotionCampaignSummary` or similar.

Reply:
I like it, makes the code feel less wonky. Updated.